compiler/src/parsing/parser.dyp

```diff
@@ -27,9 +27,21 @@ let debug_print_state () =
   when_debug !last_state_printer
 
 let line_comment source loc =
-  let content = String_utils.slice ~first:1 source |> String.trim in
+  let content = String_utils.slice ~first:2 source |> String.trim in
   Line { cmt_content=content; cmt_source=source; cmt_loc=loc }
 
+let shebang_comment source loc =
+  let content = String_utils.slice ~first:2 source |> String.trim in
+  Shebang { cmt_content=content; cmt_source=source; cmt_loc=loc }
+
+let block_comment source loc =
+  let content = String_utils.slice ~first:2 ~last:(-2) source |> String.trim in
+  Block { cmt_content=content; cmt_source=source; cmt_loc=loc }
+
+let doc_comment source loc =
+  let content = String_utils.slice ~first:3 ~last:(-2) source |> String.trim in
+  Doc { cmt_content=content; cmt_source=source; cmt_loc=loc }
+
```

Comment:
Where are these used?

Reply:
They're used in the lexer.

Reply:
@ospencer Ah, I see. I think it's a little confusing, since `line_comment` and `shebang_comment` are redefined inside of the lexer, but the others are not (so I think there are two parallel scopes going on?). Do you think there is a way to make it a little more clear what name refers to what?